### PR TITLE
add adc mux control function

### PR DESCRIPTION
**Pull request by damercer**
Add control transfer cases to chage adc mux configuration reg and dac address bits

### CODE REVIEW

**Comment by damercer** [comment]
Since I could never get a working version from master source using Atmel Studio 7 on Windows, I tried to bring back this very old, out of date, slow Linux computer of mine but after attempting to up-date the software a bunch of stuff broke like the Wifi and the video monitor driver so I cant try building my changes on Linux either. Could you try building my additions to main.c on your magic Linux tool-chain to see if my changes break anything?

Thanks

Doug

**Comment by dbogdan** [comment]
Sure, we'll let you know the result tomorrow.

**Comment by damercer** [comment]
I'm sure there will be a binary available I can download and test myself.

Thanks

Doug

**Comment by dbogdan** [comment]
The .bin was uploaded on adc_mux branch (https://github.com/analogdevicesinc/m1k-fw/tree/adc_mux). Some additional changes seem to be required in order to achieve good timing.

**Comment by damercer** [comment]
Thanks. My testing on the new .bin (V 2.17) seems to work the same as master version 2.16 in the original modes. **No bad samples in channel B at all** (or channel A for that matter). Where as the version I build using Atmel Studio 7 has some bad samples sometimes in channel B. So there is something definitely different. Also the length of your bin file is much larger than mine , 1624 bytes longer.

The new bin file also works perfectly in my new 200 KSPS modes.

So, bottom line, from what I can tell based on my testing **we have success**! I will now forge ahead on digging into libsmu to add / change functions to support new mux and dac sampling modes.

Again, thank for you help building the new binary.

Doug

**Comment by dbogdan** [comment]
This one seems to be the desired merge: from damercer:master to analogdevicesinc:master

**Review by dbogdan — APPROVED** [review]
